### DISCUSSION TITLE
Group and sum by nested property [JVM]

### DISCUSSION THREAD

**Discussion by nroi** [ravendb/ravendb#15825] [Q&A] [answered]
Hi everyone,

Given a Collection with documents that contains orders grouped by category and selling date, like this:

```json
[
    {
        "date": "2023-04-01",
        "category": "fruit",
        "amounts": [
            {
                "currency": "EUR",
                "amount": 200
            },
            {
                "currency": "USD",
                "amount": 100
            }
        ]
    },
    {
        "date": "2023-04-01",
        "category": "electronics",
        "amounts": [
            {
                "currency": "EUR",
                "amount": 300
            },
            {
                "currency": "USD",
                "amount": 400
            }
        ]
    }
]
```

I want an output that is grouped by `date` only, with the amounts summed *per currency*. This is the desired output:

```json
[
    {
        "date": "2023-04-01",
        "amounts": [
            {
                "currency": "EUR",
                "amount": 500
            },
            {
                "currency": "USD",
                "amount": 500
            },
        ]
    }
]
```

I did already manage to achieve this with an Index:

map:
```
from r in docs.Orders
select new {
    date = r.date,
    amounts = r.amounts
}
```

reduce:
```
from result in results
group result by new { result.date } into g
let amountsFlat = g.SelectMany(x => x.amounts)
select new {
    date = g.Key.date,
    amounts = amountsFlat.GroupBy(x => x.currency).Select(k => new { currency = k.Key, amount = k.Sum(x => x.amount) } )
}
```

Now, my question is: Is it possible to get this output with the RavenDB JVM client directly from the Collection, without using the index? I.e., something like:

```
List<DailyOrders> orders = session.
...
.toList()
```

Where `DailyOrders` is a Java class that includes the properties `date` and a list of `amounts`.

I know the JVM client supports `groupBy` and `selectSum`, but what I haven't been able to figure out is how to achieve this un-nesting that we have in the index query with `let amountsFlat = g.SelectMany(x => x.amounts)` as well as the aggregation of a property that's nested inside the document.

Thanks in advance.

**Comment by ayende** [comment, marked as answer]
No, a query cannot produce this result, only via an index.
